Stats page: add a table, rework sections

hyanwong opened the pull request:
## Description

Addresses some of #1498 - but this is only a draft for comments

petrelharp commented:
Thanks for having a go at this. Here's how this renders, for reference:
![Screenshot from 2021-06-17 05-52-10](https://user-images.githubusercontent.com/1046249/122400088-3f298e80-cf30-11eb-8e8c-046b24989249.png)

Let's see - LD should be under "one way" not "multi way" since it's not a comparison between >=2 groups of samples... but, "multi way" is referring to [this stuff](https://tskit.dev/tskit/docs/stable/stats.html#multi-way-methods) which doesn't even apply to LD (it's not using the same API yet). So, I'm not sure if we can have the column headers?

But more generally - I'm not sure that having things classified in this summary table up top is helpful - why not just a summary table, like for instance of the methods of a TreeSequence [here](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence)?

Other changes look good.

hyanwong commented:
> Let's see - LD should be under "one way" not "multi way" since it's not a comparison between >=2 groups of samples...

Yep, sorry - I didn't pay much attention!
> but, "multi way" is referring to [this stuff](https://tskit.dev/tskit/docs/stable/stats.html#multi-way-methods) which doesn't even apply to LD (it's not using the same API yet). So, I'm not sure if we can have the column headers?

I'm not sure it's quite about the API (although I guess we eventually might want to have similar or identical APIs for multi-site and single site stats). I think it's more about the logic. You can have a stat which is defined on a single sample set, or compares between sample sets somehow, and that's true both for multi and single site measures, right?

We might want some way to indicate that the single site stuff has a general API and the multi-site stuff doesn't. But I think having both the terms multi-way and multi-site on the page means that we need some very clear way to make the distinction, and the table does it for me (and, in addition, squeezes it all onto a single screen, which I find helpful as an overview (I can get lost in a long list like the TS methods)

> But more generally - I'm not sure that having things classified in this summary table up top is helpful - why not just a summary table, like for instance of the methods of a TreeSequence [here](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence)?

See above. But I'm very welcome to changes, especially if they mean there's a one-page summary of what's available.

petrelharp commented:
It's true that this is a distinction we *could* make up top, but I don't think we *need* to (and it's feeling kinda cluttered) - I don't think we need to remind users here that LD works on more than one site, for instance.

hyanwong commented:
> It's true that this is a distinction we _could_ make up top, but I don't think we _need_ to (and it's feeling kinda cluttered) - I don't think we need to remind users here that LD works on more than one site, for instance.

Fair enough. But then I think we need some wording to make clear the distinction between the terms multi-site and multi-way (both of which we will use on the page). Otherwise I think people will think they are the same thing. Or maybe think of another terminology for one of them?

So up the top, you would prefer something like 

- Single site
  - One-way
    - allele_frequency_spectrum
    - diversity
    - segregating_sites
    - Trait correlations: trait_covariance, trait_correlation, trait_linear_mode
    - Fst
  - Multi-way
    - divergence
    - genetic_relatedness
    - Patterson's f stats: f4, f3, f2
    - Y stats: Y3, Y2
    - Tajima's D
    - GNN
- Multi-site
  - LD
  - IBD

petrelharp commented:
I think the role of the table up top is mostly as a quick reference, so people can quickly find the stat they are looking for. And I'm a bit worried that saying 'single site' up here will make people think that we're only calculating these things one SNP at a time, so I'm leaning towards a flat table. Do you think that breaking things out in this table will introduce the distinctions?

hyanwong commented:
Maybe we just need a different terminology for "single/multi site"? Here's a mock-up of the list style. Does that work better for you? At least it fits on one page, so fine by me.

<img width="725" alt="Screenshot 2021-06-17 at 16 39 47" src="https://user-images.githubusercontent.com/4699014/122429371-a3793c00-cf8a-11eb-9569-56f0643788db.png">

petrelharp commented:
Heck, that looks good. Maybe add an explanatory sentence like "Single-site statistics are ones that are averages across sites in windows of the genome."

hyanwong commented:
Updated the PR to the list for and added @petrelharp 's suggestion.

Note that I include GNN here although it isn't in the stats framework yet. It definitely is a single site, multi-way statistic, but it explicitly ignored branch lengths. I guess we could simply explain that in a section, a bit like the one labelled "Y statistics".

It also has a time-windowing feature, which we don't discuss in this documentation (although is that fully pushed now @awohns ?)

petrelharp commented:
> Note that I include GNN here although it isn't in the stats framework yet. 

I think we were previously not documenting GNN because we were aiming to change the interface to match the general framework? This would be nice to get out of the way.

awohns commented:
A python version of windowed GNN is in the test suite (with tests),  but not the main API yet. This is something I had on the back burner but I will now make a priority since it looks like it would be a good time to incorporate it into the stats framework as @hyanwong and @petrelharp say.

petrelharp commented:
that'd be great!

hyanwong commented:
Yes, it would be a good time @awohns - thanks if you can do it!

awohns commented:
Hi @hyanwong and @petrelharp, I've become swamped with other commitments for the next month so won't be able to address the time windows (#1237) for a few weeks. Please feel free to go ahead on this without time-windows for now! (or if you want to take a crack it, I can show you what I've done with the time windows so far!)

jeromekelleher commented:
Looks like there's some good updates here but we got a bit sidetracked - could we finish this up and log issues for anything that can't be done immediately please @hyanwong?
